src/commands/azureServiceOperators/ui/azureservicebrowser.ts

```diff
@@ -0,0 +1,117 @@
+// import { TreeItem, TreeItemCollapsibleState } from 'vscode';
+import { TreeItem, TreeItemCollapsibleState } from 'vscode';
+import * as k8s from 'vscode-kubernetes-tools-api';
+import * as vscode from 'vscode';
+
+ export class AzureServiceBrowser implements k8s.ClusterExplorerV1.NodeContributor {
+    constructor(private readonly explorer: k8s.ClusterExplorerV1) {}
+    contributesChildren(parent: k8s.ClusterExplorerV1.ClusterExplorerNode | undefined): boolean {
+        return !!parent && parent.nodeType === 'context';
+    }
+    async getChildren(parent: k8s.ClusterExplorerV1.ClusterExplorerNode | undefined): Promise<k8s.ClusterExplorerV1.Node[]> {
+        if (this.contributesChildren(parent)) {
+            return [new AzureServicesFolderNode(this.explorer)];
+        }
+        return [];
+    }
+}
+
+ class AzureServicesFolderNode implements k8s.ClusterExplorerV1.Node {
+    constructor(private readonly explorer: k8s.ClusterExplorerV1) {}
+    async getChildren(): Promise<k8s.ClusterExplorerV1.Node[]> {
+        const pinnedNodes = (await pinnedServiceKinds()).map((k) => new PinnedAzureServiceKindNode(this.explorer, k));
+        const allNode = new AllAzureServicesFolderNode(this.explorer);
+        return [...pinnedNodes, allNode];
+    }
+    getTreeItem(): TreeItem {
+        const treeItem = new TreeItem("Azure Services", TreeItemCollapsibleState.Collapsed);
+        treeItem.contextValue = "aks.aso.azureservices";
+        return treeItem;
+    }
+}
+
+ class AllAzureServicesFolderNode implements k8s.ClusterExplorerV1.Node {
+    constructor(private readonly explorer: k8s.ClusterExplorerV1) {}
+    async getChildren(): Promise<k8s.ClusterExplorerV1.Node[]> {
+        const all = await allServiceKinds();
+
+        return all!.map((k) => new AzureServiceKindNode(this.explorer, k));
+    }
+    getTreeItem(): TreeItem {
+        const treeItem = new TreeItem("All Service Types", TreeItemCollapsibleState.Collapsed);
+        treeItem.contextValue = "aks.aso.allservices";
+        return treeItem;
+    }
+
+ }
+
+ class AzureServiceKindNode  implements k8s.ClusterExplorerV1.Node {
+    constructor(private readonly explorer: k8s.ClusterExplorerV1, private readonly kind: AzureServiceKind) {}
+    async getChildren(): Promise<k8s.ClusterExplorerV1.Node[]> {
+        const allFolderChildren = this.explorer.nodeSources.resourceFolder(this.kind.displayName, this.kind.displayName, this.kind.manifestKind, this.kind.abbreviation);
```

Comment:
This appears to be nesting a resource folder instead each custom folder.  Won't this give you an appearance of

```
|- context
   |- Azure Services
      |- EventHub
         |- EventHub
      |- SQLServer
         |- SQLServer
```

A resource folder _is_ a folder - not the _contents_ of a folder.